docs: improve dependencyDashboardApproval section (#8841)
renovatebot/renovate: docs/usage/configuration-options.md

## dependencyDashboardApproval

This feature allows you to use Renovate's Dependency Dashboard to force approval of updates before they are created.

By setting `dependencyDashboardApproval` to `true` in config (including within `packageRules`), you can tell Renovate to wait for your approval from the Dependency Dashboard before creating a branch/PR.
You can approve a pending PR by ticking the checkbox in the Dependency Dashboard issue.

Note: When you set `dependencyDashboardApproval` to `true` the Dependency Dashboard issue will be created automatically, you do not need to turn on `dependencyDashboard` explictly.

You can configure Renovate to wait for approval for:

- all package upgrades
- major, minor, patch level upgrades
- specific package upgrades
- upgrades coming from specific package managers

If you want to approve _all_ upgrades, set `dependencyDashboardApproval` to `true`:

```json
{
  "dependencyDashboardApproval": true
}
```

If you want to require approval for _major_ updates, set `dependencyDashboardApproval` to `true` within a `major` object:

```json
{
  "major": {
    "dependencyDashboardApproval": true
  }
}
```

> If you want to approve a _specific_ package, set `dependencyDashboardApproval` to `true` within a `packageRules` entry where you have defined a specific package.

Reviewer: ```suggestion
If you want to approve _specific_ packages, set `dependencyDashboardApproval` to `true` within a `packageRules` entry where you have defined a specific package or pattern.
```

## dependencyDashboardApproval

> By default Renovate creates branches/PRs automatically whenever a package gets an update.

Reviewer: ```suggestion
This feature allows you to use Renovate's Dependency Dashboard to force approval of updates before they are created.
```

> You can override the default behavior by setting `dependencyDashboardApproval` to `true`.
> Renovate waits for your approval from the Dependency Dashboard before creating a PR.

Reviewer: ```suggestion
By setting `dependencyDashboardApproval` to `true` in config (including within `packageRules`), you can tell Renovate to wait for your approval from the Dependency Dashboard before creating a branch/PR.
```

You can approve a pending PR by ticking the checkbox in the Dependency Dashboard issue.

> Note: When you set `dependencyDashboardApproval` to `true` the Dependency Dashboard issue will be created automatically, you do not need to turn on the `dependencyDashboardApproval` explictly.

Author: Maybe we should tell readers to turn on Dependency Dashboard as well in the config when they are using `dependencyDashboardApproval`? This will probably prevent a bunch of edge-cases?

If we want to tell them to turn on the Dashboard explicitly, I'll also change the JSON examples below to include turning on the dashboard explicitly.

Reviewer: This sentence doesn't seem to make sense right now. Did you mean they didn't need to turn on dependencyDashboard explicitly ?

Author: Whoops, I was getting my `dependencyDashboardApproval` and the plain `dependencyDashboard` mixed up.
Pushed a commit to fix the nonsensical sentence. 😄

You can configure Renovate to wait for approval for:

- all package upgrades
- major, minor, patch level upgrades
- specific package upgrades
- upgrades coming from specific package managers

If you want to approve _all_ upgrades, set `dependencyDashboardApproval` to `true`:

```json
{
  "dependencyDashboardApproval": true
}
```

> If you want to approve _major_ upgrades, set `dependencyDashboardApproval` to `true` within a `major` object:

Reviewer: ```suggestion
If you want to require approval for _major_ updates, set `dependencyDashboardApproval` to `true` within a `major` object:
```

> If you want to approve _major_ upgrades, set `dependencyDashboardApproval` to `true` within a `major` object:

Author: Not sure if this is the correct wording here for the "`major` object".

```json
{
  "major": {
    "dependencyDashboardApproval": true
  }
}
```

> If you want to approve a _specific_ package, set `dependencyDashboardApproval` to `true` within a `packageRules` array where you have defined a specific package.

Reviewer: ```suggestion
If you want to approve a _specific_ package, set `dependencyDashboardApproval` to `true` within a `packageRules` entry where you have defined a specific package.
```

```json
{
  "packageRules": [
    {

>       "matchPackagePatterns": ["^@package-name"],

Author: Should we use something that's not grouping packages together here?

Reviewer: matching patterns does not cause grouping. this is fine

      "dependencyDashboardApproval": true
    }
  ]
}
```

> If you want to approve all upgrades coming from a specific package manager, set `dependencyDashboardApproval` to `true` within a "manager" object:

Author: Should we link to the document that describes the allowed package managers?

https://docs.renovatebot.com/modules/manager/

> If you want to approve all upgrades coming from a specific package manager, set `dependencyDashboardApproval` to `true` within a "manager" object:
> 
> ```json
> {
>   "docker": {
>     "dependencyDashboardApproval": true
>   }
> }
> ```

Reviewer: ```suggestion
```

> ```

Author: Also would this even work for _all_ managers?
I understand there are some differences in configuration between managers?
Maybe we should not use this example if you cannot replace "docker" with every possible manager?

Reviewer: None of the stuff above is specific to `dependencyDashboardApproval` - what is being described is just common/generic configuration. Let's not expand it any more.

Author: Okay, I was just worried that copy/pasting in another manager name would cause troubles. 😄 

Let's keep it as-is then. 😉
